### ISSUE TITLE
Template variable saved in inconsistent state prevents selecting desired value

### ISSUE THREAD

**Issue by timpalpant** [grafana/grafana#26242]
**What happened**: A template variable selection was saved in an inconsistent state in the dashboard model, which then makes it seem unselectable on reload.

The "current" value saved in the dashboard model is inconsistent with the "selected" option:
![template-var-state](https://user-images.githubusercontent.com/814226/87169392-e96b5180-c29d-11ea-9fc7-92e54bfa9ab2.png)
![template-var-options-state](https://user-images.githubusercontent.com/814226/87169396-eb351500-c29d-11ea-97b2-8d8961d14c07.png)

In the UI this looks like:
![template-var](https://user-images.githubusercontent.com/814226/87169400-ed976f00-c29d-11ea-9ebd-7de5289ec29c.png)
![template-var-options](https://user-images.githubusercontent.com/814226/87169407-ef613280-c29d-11ea-84e9-33fb850ad697.png)

One cannot select the desired value since in the option list it is already marked selected.

**What you expected to happen**: The dashboard model was saved in a consistent way so that the desired value could be selected.

**How to reproduce it (as minimally and precisely as possible)**: I am not sure how the dashboard model got in this state, but once it is persisted this way it does not self correct.

**Anything else we need to know?**: Worked around by setting variable refresh to "On Dashboard Load"

**Environment**:
- Grafana version: 7.0.3 (dashboard was saved with 6.7.3)
- Data source type & version: custom
- OS Grafana is installed on: Ubuntu 16.04
- User OS & Browser: Safari 13.1.1
- Grafana plugins:
- Others:

**Comment by aocenas** [comment]
Thanks for reporting. Could be some migration in 7.0 that hit some edge case. From you comments though I get that this has an easy workaround right now is that right?

**Comment by timpalpant** [comment]
Yes, it was easy enough to work around by switching to another value and re-saving the dashboard.

**Comment by timpalpant** [comment]
Just noting that we have also seen this occur with dashboards edited only in 7.0, so it is unlikely to be due to the migration but rather some other race when saving dashboard state.

**Comment by timpalpant** [comment]
Hi @ashharrison90, I have not yet upgraded to Grafana 8.  But since this seems to be a rare issue that others have not run into, it is fine to close.

**Comment by ashharrison90** [comment]
enjoy Grafana 8 when you do upgrade 🥳
